insights/config/__init__.py

```diff
@@ -28,6 +28,21 @@ def __init__(self, expression, message):
         self.expression = expression
         self.message = message
 
+    def __str__(self):
+        return u"Error in path '{s.expression}': {s.message}".format(s=self)
+
+
+class SpecSyntaxError(Exception):
+    """
+        Syntactic errors in the spec path, such as use of invalid characters.
+    """
+    def __init__(self, path, message):
+        self.path = path
+        self.message = message
+
+    def __str__(self):
+        return u"Error in path '{s.path}': {s.message}".format(s=self)
```

Comment:
@PaulWay not trying to be picky here, but is `path` the correct term here?  Should it be `cmd` instead since it represents invalid characters in the bash command?

Reply:
I'm leaving the option open for syntax errors in file paths (e.g. $HOME or $MAILTO or a path starting with ~, which won't get interpolated).

Reply:
Ok by me then.